kratos/utilities/time_discretization.h

```diff
@@ -25,47 +26,124 @@ class ProcessInfo; // forward-declaring to not having to include it here
 
 namespace TimeDiscretization {
 
-class KRATOS_API(KRATOS_CORE) BDF1
+class KRATOS_API(KRATOS_CORE) BDF
 {
 public:
-    std::array<double, 2> ComputeBDFCoefficients(double DeltaTime) const;
-    std::array<double, 2> ComputeBDFCoefficients(const ProcessInfo& rProcessInfo) const;
+
+    KRATOS_CLASS_POINTER_DEFINITION(BDF);
+
+    /**
+     * @brief Construct a new BDF object
+     * Auxiliary constructor for derived classes
+     */
+    BDF(){};
```

Comment:
I would not allow this, the user should always give a time-order, since this is a "wrapper"

Reply:
This is needed for derived classes. We can move it to a `protected` section.

Reply:
Moved to a `protected`section.